view/perspective/utils.js

```diff
@@ -347,49 +344,64 @@ function getValuesObject(accumulatorObject) {
 
     /*
      * One out of four situations can happen:
+     * GET default perspective: exists. Redirect to the default perspective
+     * GET default perspective: does NOT exist: perspective is the first
+     *  perspective in the perspectives array.
+     * If no perspectives exist, valuesObj.perspective = null
+     */
+     if (!named) {
+      if (returnedPerspective) {
+
+        // the value field has the name of the default perspective.
+        // need the return statement to skip executing the rest of code
+        return redirectToUrl('/perspectives/' + returnedPerspective.value);
+      }
+
+      // default perspective does NOT exist.
+      // GET the first perspective by alphabetical order.
+      // Check to see there are perspectives
+      if (valuesObj.perspectives.length) {
+
+        // redirect to the first perspective. The rest of the code
+        // won't be executed.
+        return redirectToUrl('/perspectives/' + valuesObj.perspectives[0].name);
```

Comment:
Check for valuesObj.perspectives[0].name. Always a good practice to check for attribute before using it.